config/nav.yml

```
@@ -286,6 +286,7 @@ nav:
         - KReference.Group field: eventing/experimental-features/kreference-group.md
         - Knative reference mapping: eventing/experimental-features/kreference-mapping.md
         - EventType auto creation: eventing/experimental-features/eventtype-auto-creation.md
+        - Eventing with Istio: eventing/experimental-features/istio-integration.md
         - Transport Encryption: eventing/experimental-features/transport-encryption.md
         - Sender Identity: eventing/experimental-features/sender-identity.md
```

nit: let's keep the order in the nav the same as the order in the table of experimental features
```suggestion
        - Transport Encryption: eventing/experimental-features/transport-encryption.md
        - Sender Identity: eventing/experimental-features/sender-identity.md
        - Eventing with Istio: eventing/experimental-features/istio-integration.md
```